posthog/api/capture.py

```diff
@@ -239,11 +237,27 @@ def get_event(request):
             continue
 
         statsd.incr("posthog_cloud_plugin_server_ingestion")
-        capture_internal(event, distinct_id, ip, site_url, now, sent_at, team.pk, event_uuid)  # type: ignore
+        try:
+            capture_internal(event, distinct_id, ip, site_url, now, sent_at, team.pk, event_uuid)  # type: ignore
+        except Exception as e:
+            capture_exception(e, {"data": data})
+            statsd.incr(
+                "posthog_cloud_raw_endpoint_failure", tags={"endpoint": "capture",},
+            )
+            return cors_response(
+                request,
+                generate_exception_response(
+                    "capture",
+                    "Unable to store event. Please try again. If you are the owner of this app you can check the logs for further details.",
+                    code="server_error",
+                    type="server_error",
+                    status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+                ),
+            )
 
     timer.stop()
```

Comment:
We should stop the timer in both cases.

Reply:
Done! Not sure finally works best here as we have a return statement above.